MantidPlot/src/Mantid/MantidMatrix.cpp

```diff
@@ -531,30 +531,30 @@ void MantidMatrix::goToColumn(int col) {
 
 double MantidMatrix::dataX(int row, int col) const {
   if (!m_workspace || row >= numRows() ||
-      col >= static_cast<int>(m_workspace->readX(row + m_startRow).size()))
+      col >= static_cast<int>(m_workspace->x(row + m_startRow).size()))
```

Comment:
Could we get a reference to `x` and use that instead of asking the workspace twice for it here and in line 536?